src/main/scala/cromwell/engine/EngineFunctions.scala

```diff
@@ -1,15 +1,24 @@
 package cromwell.engine
 
 import cromwell.binding.WdlFunctions
-import cromwell.binding.values.{WdlFile, WdlInteger, WdlString, WdlValue}
+import cromwell.binding.values._
 
-import scala.util.Try
+import scala.util.{Failure, Try}
 
 trait EngineFunctions extends WdlFunctions {
   protected def read_int(params: Seq[Try[WdlValue]]): Try[WdlInteger]
   protected def read_string(params: Seq[Try[WdlValue]]): Try[WdlString]
-  protected def stdout(params: Seq[Try[WdlValue]]): Try[WdlFile]
-  protected def stderr(params: Seq[Try[WdlValue]]): Try[WdlFile]
+  protected def stdout(params: Seq[Try[WdlValue]]): Try[WdlFileLike]
+  protected def stderr(params: Seq[Try[WdlValue]]): Try[WdlFileLike]
+
+  /**
+   * Extract a single `WdlValue` from the specified `Seq`, returning `Failure` if the parameters
+   * represent something other than a single `WdlValue`.
+   */
+  protected def extractSingleArgument(params: Seq[Try[WdlValue]]): Try[WdlValue] = {
```

Comment:
Oh sorry - I read that as length `> 1`, I get it now.

We have this pattern in a lot of places and we really should find a better way to handle it, but now is not the time. Disregard.